File: apps/api/AssetManagement.Api/Models/AssetType.cs
namespace AssetManagement.Api.Models;

public class AssetType
{
    public Guid Id { get; set; }
    public required string Name { get; set; }
    public string? Description { get; set; }
    public bool IsArchived { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Asset> Assets { get; set; } = [];
    public ICollection<CustomFieldDefinition> CustomFieldDefinitions { get; set; } = [];
}

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/model/AssetType.kt
package com.assetmanagement.api.model

import jakarta.persistence.*
import java.time.Instant
import java.util.*

@Entity
@Table(name = "asset_types")
class AssetType(
    @Id
    @Column(name = "id", columnDefinition = "CHAR(36)")
    var id: UUID = UUID.randomUUID(),

    @Column(name = "name", nullable = false)
    var name: String = "",

    @Column(name = "description")
    var description: String? = null,

    @Column(name = "is_archived", nullable = false)
    var isArchived: Boolean = false,

    @Column(name = "created_at", nullable = false)
    var createdAt: Instant = Instant.now(),

    @Column(name = "default_depreciation_months")
    var defaultDepreciationMonths: Int? = null,

    @Column(name = "updated_at", nullable = false)
    var updatedAt: Instant = Instant.now(),

    @OneToMany(mappedBy = "assetType")
    var assets: MutableList<Asset> = mutableListOf(),

    @OneToMany(mappedBy = "assetType", cascade = [CascadeType.ALL], orphanRemoval = true)
    var customFieldDefinitions: MutableList<CustomFieldDefinition> = mutableListOf()
)
